### DISCUSSION TITLE
Cronicle Scheduler that is version controlled

### DISCUSSION THREAD

**Discussion by correctcareak** [jhuckaby/Cronicle#745] [Q&A]
Hello Team,

I just started using Croncile and I love what it offer. Thank you for developing this. I have a question. I am trying to simplify our deployment processes. Does cronicle support a way for all the schedules to be version controlled. Meaning the schedules to be written as a line of code and then deployed to the server. I have about 100-200 scripts that run at any given time, and in addition we are setting up a three system structure. dev/stage/prod. and creating all those schedules manually on each server is time consuming. But if I had a way to automatically create them by automatically deploying to the server plus have it version controlled would be a big time saver.

Thanks for you input and ideas around my dilemma.

**Comment by jhuckaby** [comment]
Check out the data import / export guide: https://github.com/jhuckaby/Cronicle/blob/master/docs/CommandLine.md#data-import-and-export

This should do what you want, possibly with a little bit of tweaking inside of the exported file (for e.g. remove all the lines except the ones containing `global/schedule`).

**Reply by correctcareak** [reply]
Thanks, I've thought about this import option. It seems like in order to use the import function, all schedules are in one line and trying to read that it makes it more complicated, unless another script is created that is able to parse it into a format that is required by the import function.
Would I be able to directly make the source code changes on the schedules itself without needing to use the import function?

**Reply by jhuckaby** [reply]
Well, there's also the API.  You could use the [get_schedule](https://github.com/jhuckaby/Cronicle/blob/master/docs/APIReference.md#get_schedule) call to fetch the schedule, and then use [create_event](https://github.com/jhuckaby/Cronicle/blob/master/docs/APIReference.md#create_event) to recreate the events on another environment.
